### PR TITLE
Add menu components

### PR DESCRIPTION
**Pull request by jpzwarte**

### CODE REVIEW

File: packages/components/menu/src/menu.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+import { MenuItemGroup } from './menu-item-group.js';
+import { MenuItem } from './menu-item.js';
+
+export class Menu extends LitElement {
+  /** @private */
+  static override styles: CSSResultGroup = styles;
+
+  /** Controller for managing anchoring. */
+  #anchor = new AnchorController(this, {
+    offset: 4,
```
Comment (anna-lach) [inline packages/components/menu/src/menu.ts]: Can you please also add the offset for tooltip, popover and tabs (listbox), everywhere should be `offset 4` as well :)

image

File: packages/components/menu/src/menu-item.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+import { type ScopedElementsMap, ScopedElementsMixin } from '@open-wc/scoped-elements/lit-element.js';
+import { Icon } from '@sl-design-system/icon';
+import { type EventEmitter, EventsController, ShortcutController, event } from '@sl-design-system/shared';
+import { type CSSResultGroup, LitElement, type PropertyValues, type TemplateResult, html, nothing } from 'lit';
+import { property, state } from 'lit/decorators.js';
+import { Menu } from './menu.js';
+import styles from './menu-item.scss.js';
+
+Icon.register(faCheck, faChevronRight);
+
```
Comment (anna-lach) [inline packages/components/menu/src/menu-item.ts:11]: Can you please add the documentation part here about slots?

File: packages/components/menu/src/menu-item-group.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import { type CSSResultGroup, LitElement, type TemplateResult, html } from 'lit';
+import { property } from 'lit/decorators.js';
+import styles from './menu-item-group.scss.js';
+import { MenuItem } from './menu-item.js';
+
```
Comment (anna-lach) [inline packages/components/menu/src/menu-item-group.ts:5]: Can you please add the documentation part here about slots?

File: packages/components/menu/src/menu-button.ts [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+import { type ScopedElementsMap, ScopedElementsMixin } from '@open-wc/scoped-elements/lit-element.js';
+import { Button, type ButtonFill, type ButtonSize, type ButtonVariant } from '@sl-design-system/button';
+import { Icon } from '@sl-design-system/icon';
+import { type CSSResultGroup, LitElement, type PropertyValues, type TemplateResult, html, nothing } from 'lit';
+import { property, query, state } from 'lit/decorators.js';
+import { Menu } from './menu.js';
+import styles from './menu-button.scss.js';
+
+Icon.register(faAngleDown);
+
```
Comment (anna-lach) [inline packages/components/menu/src/menu-button.ts:13]: Can you please add the documentation part here about slots?

File: packages/components/shared/src/controllers/anchor.ts [hunk 5]
```diff
@@ -29,66 +20,53 @@ export class AnchorController implements ReactiveController {
...
-
-    let anchorElement = host.anchorElement;
-    if (!anchorElement && host.hasAttribute('anchor')) {
-      anchorElement =
-        (host.getRootNode() as HTMLElement)?.querySelector(`#${host.getAttribute('anchor') ?? ''}`) || undefined;
-    }
-
-    if (host.matches('sl-popover')) {
-      anchorElement?.setAttribute('aria-expanded', 'false');
-      anchorElement?.setAttribute('aria-controls', host.id);
```
[the remark this answers is not in the document]
Comment (anna-lach) [reply packages/components/shared/src/controllers/anchor.ts:89]: And maybe for the menu button there should be also `aria-expanded`? https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/
Right now VoiceOver reads the menu button just like a normal button and there is no information that sth is connected to it?

File: packages/components/menu/src/menu-item.scss [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+:host(:hover) {
+  position: relative;
+
+  .safe-triangle {
+    display: block;
+  }
+}
+
+:host(:focus-visible) {
+  outline: var(--_focus-outline);
```
Comment (anna-lach) [inline packages/components/menu/src/menu-item.scss:28]: Outline and hover next to each other looks a little bit weird. I had similar problem in tabs and that's why I manipulated with higher z-index there.


image

File: packages/components/shared/src/controllers/anchor.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (jpzwarte) [reply packages/components/shared/src/controllers/anchor.ts:89]: I just updated this code to add `aria-expanded` and `aria-details` for all popovers that use the `AnchorController`.

File: packages/components/menu/src/menu-item.scss (hunk 6 above)
Comment (jpzwarte) [reply packages/components/menu/src/menu-item.scss:28]: Fixed

File: packages/components/tooltip/src/tooltip.ts [hunk 7]
```diff
@@ -38,11 +44,8 @@ export class Tooltip extends LitElement {
     ['focusin', 'pointerover'].forEach(eventName => target.addEventListener(eventName, createTooltip));
   }
 
-  /** Tooltip max-width. */
-  @property({ type: Number, attribute: 'max-width' }) maxWidth?: number;
-
   /** Controller for managing anchoring. */
-  #anchor = new AnchorController(this, { maxWidth: this.maxWidth });
+  #anchor = new AnchorController(this, { offset: Tooltip.offset, viewportMargin: Tooltip.viewportMargin });
```
[the remark this answers is not in the document]
Comment (jpzwarte) [reply packages/components/tooltip/src/tooltip.ts:49]: This is already broken on `main`, so I'd like to fix this in a separate branch.

**Review by anna-lach — APPROVED** [review]

**Pull request merged**